ypfs.js

```diff
@@ -0,0 +1,223 @@
+{
+	"translatorID": "5e1f3e08-ca5f-4196-a662-6cf46b3cdaa7",
+	"label": "YPFS",
+	"creator": "Corey Runkel",
+	"target": "^https?://ypfs\\.som\\.yale\\.edu",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-03-24 13:06:04"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2021 Corey Runkel
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (url.includes('/library/')) {
+		return 'document';
+
+	} else if (url.endsWith('.edu') || url.endsWith('.edu/') || url.includes('admin/content')) {
+		return 'multiple';
+	}
+}
+
+
+ function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('a[href*="/library/"]');
+	for (let row of rows) {
+		href = row.href;
+		title = ZU.trimInternal(row.text);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var type = itemType(doc.querySelector('a[href*="ypfsresourcelibrary"]').href);
```

Comment:
Is there another selector that could work here? It feels somewhat likely that the first link on the page matching that string won't *always* be the one you want here.

Reply:
At the risk of sounding arrogant, I am confident that a matching string will always be the one users want. `detectWeb` significantly limits the type of pages that `scrape` can be used on